wallet/walletrpc.c

```diff
@@ -390,23 +391,47 @@ static const struct json_command newaddr_command = {
 };
 AUTODATA(json_command, &newaddr_command);
 
-static void json_listfunds(struct command *cmd, const char *buffer,
-			   const jsmntok_t *params)
+static void json_listfunds(struct command *cmd, const char *buffer UNUSED,
+			   const jsmntok_t *params UNUSED)
 {
 	struct json_result *response = new_json_result(cmd);
+	struct peer *p;
 	struct utxo **utxos =
 	    wallet_get_utxos(cmd, cmd->ld->wallet, output_state_available);
 	json_object_start(response, NULL);
 	json_array_start(response, "outputs");
-	for (int i = 0; i < tal_count(utxos); i++) {
+	for (size_t i = 0; i < tal_count(utxos); i++) {
 		json_object_start(response, NULL);
 		json_add_txid(response, "txid", &utxos[i]->txid);
 		json_add_num(response, "output", utxos[i]->outnum);
 		json_add_u64(response, "value", utxos[i]->amount);
 		json_object_end(response);
 	}
 	json_array_end(response);
+
+	/* Add funds that are allocated to channels */
+	json_array_start(response, "channels");
+	list_for_each(&cmd->ld->peers, p, list) {
+		if (!p->our_msatoshi || !p->funding_txid)
+			continue;
+
+		json_object_start(response, NULL);
+		json_add_pubkey(response, "peer_id", &p->id);
+		if (p->scid)
+			json_add_short_channel_id(response, "short_channel_id",
+						  p->scid);
+		else
+			json_add_null(response, "short_channel_id");
+
+		json_add_u64(response, "value_sat", *p->our_msatoshi/1000);
```

Comment:
Somewhat ambivalent, this rounds down so if I have two channels, with 1500msat on one and 1500 on other, I have 3 satoshi total, but this will report 1 satoshi it (summing to 2 satoshi).  But, utxo report uses satoshi unit, so also confusing if we switch to msatoshi here.

Reply:
How about if we round "correctly"? `(*p->our_msatoshi + 500)/1000` is a cheap way to do that, summing on both ends still won't work, but it's correct(ish)